Sources/JWTKit/RSA/PrimeGenerator.swift

```diff
@@ -0,0 +1,130 @@
+import Foundation
+
+struct PrimeGenerator {
+    /// The following algorithm recovers the prime factors of a modulus, given the public and private exponents. 
+    /// The algorithm is based on Fact 1 in [Boneh 1999].
+    public static func calculatePrimeFactors(n: BigInt, e: BigInt, d: BigInt) throws -> (p: BigInt, q: BigInt) {
+        let k = (d * e) - 1
+
+        guard k & 1 == 0 else {
+            throw RSAError.keyInitializationFailure
+        }
+
+        var r = k
+        var t = 0
+
+        repeat {
+            r = r / 2
+            t &+= 1
+        } while r.words[0] & 1 == 0
```

Comment:
I just realized you don't even need a loop for this 😅 
```suggestion
        let t = k.trailingZeroBitCount, r = k >> t
```